src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheckTest.java

```diff
@@ -311,6 +314,69 @@ public void testWrongSingletonTagInJavadoc() throws Exception {
                 expected);
     }
 
+    @Test
+    public void testNonTightHtmlTagIntolerantCheck() throws Exception {
+        NonTightHtmlTagIntolerantCheck.visitJavadocTokenMethodCallCount = 0;
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(NonTightHtmlTagIntolerantCheck.class);
+        checkConfig.addAttribute("violateExecutionOnNonTightHtml", "true");
+        final String[] expected = {
+            "6: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+            "13: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+            "16: " + getCheckMessage(MSG_UNCLOSED_HTML, "li"),
+            "21: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+            "27: " + getCheckMessage(MSG_UNCLOSED_HTML, "tr"),
+            "34: " + getCheckMessage(MSG_UNCLOSED_HTML, "li"),
+            "54: " + getCheckMessage(MSG_UNCLOSED_HTML, "li"),
+            "71: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+            "80: " + getCheckMessage(MSG_UNCLOSED_HTML, "tr"),
+            "124: " + getCheckMessage(MSG_UNCLOSED_HTML, "p"),
+        };
+        verify(checkConfig, getPath("InputAbstractJavadocNonTightHtmlTags.java"), expected);
+    }
+
+    @Test
+    public void testNonTightHtmlTagIntolerantCheckReportingNoViolation() throws Exception {
+        NonTightHtmlTagIntolerantCheck.visitJavadocTokenMethodCallCount = 0;
+        final DefaultConfiguration checkConfig =
+                createModuleConfig(NonTightHtmlTagIntolerantCheck.class);
+        final String[] expected = CommonUtils.EMPTY_STRING_ARRAY;
+        verify(checkConfig, getPath("InputAbstractJavadocNonTightHtmlTags.java"), expected);
+    }
+
+    @Test
+    public void testNonTightHtmlTagIntolerantCheckVisitCount()
+            throws Exception {
+        NonTightHtmlTagIntolerantCheck.visitJavadocTokenMethodCallCount = 0;
```

Comment:
Wouldn't it make more sense for these resets to be in the constructor of `NonTightHtmlTagIntolerantCheck`?

Reply:
@rnveach PMD violation about assignment of `non-final` static fields in constructor.

Reply:
@ps-sp I see, just revert the code unless you see a way to fix it. I thought we had other examples of doing similar things.
Please run verify before pushing new changes to make sure it passes. Even I don't know all the checks we have enabled.

Reply:
@ps-sp You could do the reset in `beginTree` instead, right?

Reply:
`beginTree` is final in `AbstractJavadocCheck` and `beginJavadocTree` is called for each javadoc token to be processed. Neither can be used apparently.

Reply:
Reverted. 

An alternative that I can think of is use a static method for clearing the static variable which doesn't seem to do any good. So presently I feel that the current approach is the most efficient one.